x/auth/client/cli/cli_test.go

```diff
@@ -1024,6 +1024,85 @@ func (s *IntegrationTestSuite) createBankMsg(val *network.Validator, toAddr sdk.
 	return res
 }
 
+func (s *IntegrationTestSuite) TestSignWithMultiSigners() {
+	val0, val1 := s.network.Validators[0], s.network.Validators[1]
+	val0Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val0.Moniker), sdk.NewInt(10))
+	val1Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val1.Moniker), sdk.NewInt(10))
+	_, _, addr1 := testdata.KeyTestPubAddr()
+
+	// Creating a tx with 2 msgs from 2 signers: val0 and val1.
+	txBuilder := val0.ClientCtx.TxConfig.NewTxBuilder()
+	txBuilder.SetMsgs(
+		banktypes.NewMsgSend(val0.Address, addr1, sdk.NewCoins(val0Coin)),
+		banktypes.NewMsgSend(val1.Address, addr1, sdk.NewCoins(val1Coin)),
+	)
+	txBuilder.SetFeeAmount(sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))))
+	txBuilder.SetGasLimit(testdata.NewTestGasLimit())
+	// Set signer_infos for both signers. Note: we use the empty signature hack.
+	txBuilder.SetSignatures(
+		signing.SignatureV2{
+			PubKey:   val0.PubKey,
+			Data:     &signing.SingleSignatureData{SignMode: signing.SignMode_SIGN_MODE_DIRECT},
+			Sequence: 45, // TODO Replace with real seq, this should fail in antehandler
+		},
+		signing.SignatureV2{
+			PubKey:   val1.PubKey,
+			Data:     &signing.SingleSignatureData{SignMode: signing.SignMode_SIGN_MODE_DIRECT},
+			Sequence: 67, // TODO Replace with real seq, this should fail in antehandler
+		},
+	)
+	s.Require().Equal([]sdk.AccAddress{val0.Address, val1.Address}, txBuilder.GetTx().GetSigners())
+
+	// Write the unsigned tx into a file.
+	txJSON, err := val0.ClientCtx.TxConfig.TxJSONEncoder()(txBuilder.GetTx())
+	s.Require().NoError(err)
+	unsignedTxFile := testutil.WriteToNewTempFile(s.T(), string(txJSON))
+
+	// TODO Remove println
+	fmt.Println("== unsigned ==")
+	fmt.Println(string(txJSON))
+
+	// Let val0 sign first the file with the unsignedTx.
+	signedByVal0, err := authtest.TxSignExec(val0.ClientCtx, val0.Address, unsignedTxFile.Name(), "--overwrite", fmt.Sprintf("--%s=direct", flags.FlagSignMode))
+	s.Require().NoError(err)
+	signedByVal0File := testutil.WriteToNewTempFile(s.T(), signedByVal0.String())
+
+	// TODO Remove println
+	fmt.Println("== signed by val0 ==")
+	fmt.Println(signedByVal0.String())
+
+	// Then let val1 sign the file with signedByVal0.
+	val1AccNum, val1Seq, err := val0.ClientCtx.AccountRetriever.GetAccountNumberSequence(val0.ClientCtx, val1.Address)
+	s.Require().NoError(err)
+	signedTx, err := authtest.TxSignExec(
+		val1.ClientCtx, val1.Address, signedByVal0File.Name(),
+		"--offline", fmt.Sprintf("--account-number=%d", val1AccNum), fmt.Sprintf("--sequence=%d", val1Seq), fmt.Sprintf("--%s=direct", flags.FlagSignMode),
+	)
+	s.Require().NoError(err)
+	signedTxFile := testutil.WriteToNewTempFile(s.T(), signedTx.String())
```

Comment:
The most anticipated test is:
1. User1 signs and sends the signature to Relayer (that's why we have `--sign-only`)
2. User2 signs and sends to the signature to Relayer
3. Relayer aggregates the signatures in the transaction and broadcasts it.

Note: here the Relayer is a test.